src/redux/downloadevent/downloadEventMiddleware.tsx

```diff
@@ -0,0 +1,198 @@
+          notifications.update({
+            id: downloads[data.id],
+            title: i18n.t('download-event.title.finished', { ns: 'notifications' }),
+            message: (
+              <TruncatedText tooltipProps={{ zIndex: 1000 }}>
+                {i18n.t('download-event.message.finished', {
+                  ns: 'notifications',
+                  file: data.id
+                })}
+              </TruncatedText>
+            ),
+            icon: <CheckIcon size={IconSize.sm} />,
+            autoClose: true,
+            loading: false,
+            color: 'green'
+          });
+          delete downloads[data.id];
+        }
+
+        if (data.type === DownloadType.Failed) {
+          const color = 'red';
+
+          notifications.update({
+            id: downloads[data.id],
+            title: i18n.t('download-event.title.failed', { ns: 'notifications' }),
+            message: (
+              <DownloadEventNotificationBody
+                message={i18n.t('download-event.message.failed', {
+                  ns: 'notifications',
+                  file: data.id
+                })}
+                downloadProgress={50}
+                color={color}
+                animated={false}
+              />
+            ),
+            icon: <ErrorIcon size={IconSize.sm} />,
+            autoClose: true,
+            loading: false,
+            color: color
+          });
+
+          delete downloads[data.id];
+        }
+
+        if (data.type === DownloadType.Started && !downloads[data.id]) {
+          notifications.show({
+            id: data.id,
+            title: i18n.t('download-event.title.started', {
+              ns: 'notifications',
+              file: data.id
+            }),
+            message: (
+              <DownloadEventNotificationBody
+                message={i18n.t('download-event.message.started', {
+                  ns: 'notifications',
+                  file: data.id
+                })}
+                downloadProgress={0}
+              />
+            ),
+            autoClose: false,
+            loading: true,
+            color: 'white'
+          });
+          downloads[data.id] = data.id;
+        }
+
+        if (data.type === DownloadType.Progress) {
+          if (downloads[data.id]) {
+            notifications.update({
+              id: data.id,
+              title: i18n.t('download-event.title.progress', {
+                ns: 'notifications',
+                file: data.id
+              }),
+              message: (
+                <DownloadEventNotificationBody
+                  message={i18n.t('download-event.message.progress', {
+                    ns: 'notifications',
+                    file: data.id
+                  })}
+                  downloadedSize={downloadedMb}
+                  totalSize={totalMb}
+                />
+              ),
+              autoClose: false,
+              loading: true,
+              color: 'white'
```

Comment:
the contents of these functions are the same - store in variable instead of duplicating?